Documentation/ceph-pool-crd.md

```diff
@@ -84,6 +84,7 @@ When creating an erasure-coded pool, it is highly recommended to create the pool
 * `deviceClass`: Sets up the CRUSH rule for the pool to distribute data only on the specified device class. If left empty or unspecified, the pool will use the cluster's default CRUSH root, which usually distributes data over all OSDs, regardless of their class.
 * `crushRoot`: The root in the crush map to be used by the pool. If left empty or unspecified, the default root will be used. Creating a crush hierarchy for the OSDs currently requires the Rook toolbox to run the Ceph tools described [here](http://docs.ceph.com/docs/master/rados/operations/crush-map/#modifying-the-crush-map).
 * `compressionMode`: Sets up the pool for inline compression when using a Bluestore OSD. If left unspecified does not setup any compression mode for the pool. Values supported are the same as Bluestore inline compression [modes](https://docs.ceph.com/docs/master/rados/configuration/bluestore-config-ref/#inline-compression), such as `none`, `passive`, `aggressive`, and `force`.
+* `pgAutoscaleMode`: Sets the [pg_autoscale_mode](https://docs.ceph.com/docs/master/rados/operations/placement-groups/#autoscaling-placement-groups) for the given pool. Supported values are: `off`, `on`, or `warn`.
```

Comment:
Do we really need to expose this as a property on the pool CR? I'm not sure it's necessary. But I do like the idea of automatically enabling it if the pg_autoscaler module is enabled in the cluster CR.

Reply:
Currently, it's not a must-have but it's the elegant way to fix #5608 with the bonus of adding a new capability to the CephBlockPool CR. But the more I think about this the more I'm leaning towards something different.

We keep adding this one to one mapping of pool properties between the CephBlockPool CR and the Ceph pool property. It's not very practical and increases the number of parameters. I think we should have a `map[string]string{}` instead.

Let me try to change that.